libs/go/libp2p_node/dht/dhtclient/dhtclient.go

```diff
@@ -338,12 +337,14 @@ func (dhtClient *DHTClient) newStreamLoopUntilTimeout(
 			stream, err = dhtClient.routedHost.NewStream(ctx, peerID, streamType)
 		case <-ctx.Done():
 			sleepTime = 0
-			break
```

Comment:
pointless

Reply:
it's needed to exit connection loop
but you can set `err` here. so it will exit by err `for err != nil`

Reply:
we exit with line 344